Coordinate.cpp

```diff
@@ -1,11 +1,78 @@
 #include "Coordinate.h"
+#include <cmath>
 
 Coordinate::Coordinate() : x(0), y(0) {
 }
 
 Coordinate::Coordinate(int x, int y) : x(x), y(y) {
 }
 
+Coordinate::Coordinate(Direction d)
```

Comment:
I did this because we had a function to add a Direction to a cellcoordinate/nodecoordinate using a switch statement. I found that casting a direction to a coordinate allows us to remove these switch statements.